cli/xmlanalysisreport.cpp

```diff
@@ -0,0 +1,32 @@
+/*
+ * Cppcheck - A tool for static C/C++ code analysis
+ * Copyright (C) 2007-2022 Cppcheck team.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "xmlanalysisreport.h"
+
+XMLAnalysisReport::XMLAnalysisReport(const std::string& productName) {
+    mBuffer << ErrorMessage::getXMLHeader(productName) << std::endl;
```

Comment:
The buffer allows it to abstract away from its final destination, whether that's STDOUT, STDERR, or even a file (`--output-file`). I can delete the buffer, but it would mean more tight coupling to the output stream.